packages/util/src/provider.ts

```diff
@@ -27,6 +37,15 @@ export const fetchFromProvider = async (url: string, params: rpcParams) => {
     method: 'POST',
     body: data,
   })
+  if (!res.ok) {
+    throw new Error(
+      `JSONRpcError: ${JSON.stringify(
+        { method: params.method, status: res.status, message: res.text() },
```

Comment:
There is a corner case where this res.text() will throw too. This is when the network fails before the response is fully recieved in this case res.text() will throw. I think this is fine though since it's such a unique corner case

Reply:
fixed